effects.cc

```diff
@@ -201,18 +183,14 @@ void Effects_manager::remove_text_effect(
 void Effects_manager::remove_all_effects(
     bool repaint
 ) {
-	if (!effects && !texts)
+	if (!effects && texts.empty())
 		return;
 	while (effects) {
```

Comment:
Of course!
This is implied, I just prefer to do it in smaller batches, maybe even separate MR, but it's already work in progress :)

Reply:
@marzojr side remark: I wonder if `texts` actually should be a list; why not make it a set of some sort? Then checking for duplicates happens automagically.
I have some doubt related to `void add_text(const char *msg, int x, int y);` when it comes to `set` but at the same time I doubt there would be collisions.

Reply:
I thought of suggesting a set (or unordered_set) as well. The drawback is that it would change the ordering of displayed text. Currently, and with your changes, the newest text is drawn first, with each subsequent text going on top of all those before; with the set, the object address in RAM would set the ordering, making it basically random. In the end, at least for now, I think keeping the list is the better option.

Reply:
As for `!effects`, see https://github.com/alagner/exult/tree/list_specials
I will create a PR when I'm done with that, unless you want to have it all in one place?
I'm fine with both, though be warned, there's way more code in the latter case; due to that I would not recommend reviewing it both changes as a whole, but your call in the end.